### PR TITLE
Feature/api service call restructure

### PR DESCRIPTION
**Pull request by Tiewhan**
<h1>Overview</h1>
This feature revamps how the application handles service calls to the API

<h2>Changes</h2>
The API Manager has been removed and replaced with Service Caller.
Service Caller is a more generic version that can handle any request. It will attempt to make a service call with a url and the closure responses.

Decoding of the JSON responses is also now handled by the GameManager

### CODE REVIEW

File: Source/Managers/GameManager.swift [hunk 1]
```diff
@@ -26,11 +40,29 @@ public class GameManager {
...
+    let serviceCaller = ServiceCaller()
+    
+    let url: URL = URL(string: SteamURLComponents.gameListURL)!
+    
+    // MARK: Call Failed Closure
+    serviceCaller.callFailed = { error in
+      
+    }
+    
+    // MARK: ----------------------------
```
Comment (AMadanlal) [inline Source/Managers/GameManager.swift]: there is a lot of wasted space here

Comment (AMadanlal) [reply Source/Managers/GameManager.swift]: you can remove some of them to make it look better

**Comment by AMadanlal** [comment]
Apart from spaces, all looks good

**Review by AMadanlal — APPROVED** [review]

**Review by AkuaQ — APPROVED** [review]

**Review by pjwelcome — COMMENTED** [review]
The whitespace in some of the code files are weird.

**Pull request merged**